backend/inspirehep/records/api/literature.py

```diff
@@ -191,12 +201,10 @@ def add_files(self, data):
         if "figures" in builder.record:
             data["figures"] = builder.record["figures"]
 
-        if self.files:
-            for key in list(self.files.keys):
-                if key not in keys:
-                    del self.files[key]
         if "_files" in self:
             data["_files"] = copy(self["_files"])
+        if "_bucket" in self:
```

Comment:
Isn't just easier here to do:
```suggestion
        data["_bucket"] = str("self.bucket")
```
It will also create bucket if it's missing.

Reply:
I want to be sure I'm using the same bucket as the one used for adding the files hence it has updated the ``self``

Reply:
Then if `_bucket` is not in self we should throw some exception here. right?

Reply:
it will throw a ``KeyError``, but this case I don't see how it can happen